### PR TITLE
Add Ownership Tests to TSAN, Fix TSAN Lock Order Issues

### PR DESCRIPTION
**Pull request by simpsont-oci**
Problem: Investigating why modeling tests were failing, I discovered TSAN issues related to exclusive ownership.

Solution: Add Ownership tests to TSAN list and fix issues.

### CODE REVIEW

**Comment by simpsont-oci** [comment]
Obviously, this PR does not fix the modeling test issues. 😞 But it does fix an issue or two for exclusive ownership and does add TSAN test coverage, so still valid.

File: dds/DCPS/DataReaderImpl.h [hunk 1]
```diff
@@ -467,20 +467,50 @@ class OpenDDS_Dcps_Export DataReaderImpl
...
     }
 
   private:
     RcHandle<DomainParticipantImpl> participant_;
   };
   friend class OwnershipManagerPtr;
 
+  struct OwnershipManagerScopedAccess {
+    OwnershipManagerScopedAccess() : om_(0), lock_result_(0) {}
+    OwnershipManagerScopedAccess(DataReaderImpl::OwnershipManagerPtr om) : om_(om), lock_result_(om_ ? om_->instance_lock_acquire() : 0) {}
```
Comment (mitza-oci) [inline dds/DCPS/DataReaderImpl.h:486]:
```suggestion
    explicit OwnershipManagerScopedAccess(OwnershipManagerPtr om)
      : om_(om), lock_result_(om_ ? om_->instance_lock_acquire() : 0) {}
```

File: dds/DCPS/DataReaderImpl.h [hunk 2]
```diff
@@ -467,20 +467,50 @@ class OpenDDS_Dcps_Export DataReaderImpl
...
+    {
+      OwnershipManagerPtr om = om_;
+      om_ = sa.om_;
+      sa.om_ = om;
+      int lock_result = lock_result_;
+      lock_result_ = sa.lock_result_;
+      sa.lock_result_ = lock_result;
+    }
+
+    DataReaderImpl::OwnershipManagerPtr om_;
```
Comment (mitza-oci) [inline dds/DCPS/DataReaderImpl.h:510]:
```suggestion
    OwnershipManagerPtr om_;
```

File: dds/DCPS/DataReaderImpl.h [hunk 3]
```diff
@@ -467,20 +467,50 @@ class OpenDDS_Dcps_Export DataReaderImpl
...
+    }
+
+    void swap(OwnershipManagerScopedAccess& sa)
+    {
+      OwnershipManagerPtr om = om_;
+      om_ = sa.om_;
+      sa.om_ = om;
+      int lock_result = lock_result_;
+      lock_result_ = sa.lock_result_;
+      sa.lock_result_ = lock_result;
```
Comment (mitza-oci) [inline dds/DCPS/DataReaderImpl.h:507]: Could each of these sets of 3 statements be replaced by std::swap? Could the whole method be replaced by std::swap of the OwnershipManagerScopedAccess?

**Review by jrw972 — APPROVED** [review]

**Pull request merged**